osu.Game/Scoring/ScoreImporter.cs

```diff
@@ -88,6 +88,11 @@ protected override void Populate(ScoreInfo model, ArchiveReader? archive, Realm
             // this requires: max combo, statistics, max statistics (where available), and mods to already be populated on the score.
             if (StandardisedScoreMigrationTools.ShouldMigrateToNewStandardised(model))
                 model.TotalScore = StandardisedScoreMigrationTools.GetNewStandardised(model);
+            else if (model.IsLegacyScore)
+            {
+                model.LegacyTotalScore = model.TotalScore;
+                model.TotalScore = StandardisedScoreMigrationTools.ConvertFromLegacyTotalScore(model, beatmaps());
```

Comment:
Now that I look at this again - shouldn't this be scoring `model.Version` (or, `model.ScoreVersion`, if that rename happens, I guess) to `LegacyScoreEncoder.LATEST_VERSION`? And probably unconditionally in both branches of this `if` statement, too.

There's _possibly_ an argument here that it might be something that the `StandardisedScoreMigration` classes should be doing themselves, except for the fact that right now they are mostly pure methods, and if you had them set the version, they would cease to be pure. Which is also a problem when you consider that a willy-nilly write to an attached realm model instance may throw if done outside of a transaction.

Reply:
Okay never mind I guess this will happen anyway because of the default value applied to `ScoreInfo.Version`? Which feels... risky, but works, I guess.